publish_courses/api.py

```diff
@@ -159,171 +146,97 @@ def create(self, request, *args, **kwargs) -> Response:
+                                            audit_user_id, audit_user_name, selected_courses)
+        self.send_job_to_queueing_lambda(job_dict)
 
-        logger.info(
-            f'The bulk publish courses job creation is complete and has been sent to the SQS queue.')
+        logger.info(f'The bulk publish courses job creation for job ID {job_dict["job_id"]} is complete.')
         return Response(status=status.HTTP_201_CREATED)
-
-
-    def get_canvas_courses(self, account: str, term: str) -> List[Dict]:
+    
+    def _get_courses(self, account: str, term: str) -> List:
         """
-        Fetches a list of Canvas courses for a specified account and term. 
+        Returns a list of canvas courses for the given account and term.
         """
-        logger.debug(f'Retrieving Canvas courses.', extra={
-                     'account': account, 'term': term})
+        logger.info(f"Retrieving all courses for {account} and term {term}.")
 
-        canvas_courses = []
-        try:
-            canvas_courses = get_all_list_data(
-                SDK_CONTEXT,
-                list_active_courses_in_account,
-                account_id=account,
-                enrollment_term_id=term,
-                published=False,
-            )
-        except Exception as e:
-            message = f'Error getting courses for account {account}'
-            if isinstance(e, CanvasAPIError):
-                message += ', SDK error details={}'.format(e)
-            logger.exception(message)
-            raise e
+        op_config = {
+            'account': account,
+            'term': term
+        }
+        op = BulkCourseSettingsOperation(op_config)
+        op.get_canvas_courses()
 
-        return canvas_courses
+        logger.info(f"Retrieved {len(op.canvas_courses)} courses for {account} and term {term}.")
+        return op.canvas_courses
     
-    def create_and_save_job(self, job_id: str, account_sis_id: str, audit_user_id: int,
-                            audit_user_name: str, selected_courses: List[int],
-                            canvas_courses: List[Dict]) -> None:
+    def create_and_save_job(self, account_sis_id: str, term: str, audit_user_id: int,
+                            audit_user_name: str, selected_courses: List[int]) -> Dict:
         """
-        Retrieves workflow states for selected courses, creates a job record,
-        and job details then adds them to the database.
+        Creates a job record, then creates job details then adds them to the database.
+        Returns a dictionary containing job and job details (course IDs and PKs).
         """
+        logger.info(f'Saving bulk publish courses job to database.')
+        
         # Create job record in the database.
-        job = Job.objects.create(job=job_id,
-                                 school_id=account_sis_id[len('school:'):],
+        job = Job.objects.create(school_id=account_sis_id[len('school:'):],
```

Comment:
Why is len() being used here?

Reply:
I'm using it so that, upon saving, the school name is stored as `hls` instead of `school:hls`. Please let me know if we prefer to store the entire string, such as `school:hls`.

Reply:
I am not sure what the expected behavior is in a sub-sub-account. Currently it is broken when trying to access it but we may want to track the account_sis_id (school:hls, coursegroup:3435, dept:6314, etc). 
Are we using that value in any downstream calls?

If you are looking to remove the text in front of a string I would use the removeprefix function instead of slicing the string.
eg:

`account_sis_id.removeprefix("school:")`

Reply:
Sounds good! I will keep the prefix so we can track better (shool/dept etc..). And thanks for the tip on using `removeprefix()`!

We are not using that value in any downstream calls, so for now I won't slice or remove prefix and just save the value for `account_sis_id`.